src/radical/pilot/agent/scheduler/base.py

```diff
@@ -790,6 +783,22 @@ def _schedule_waitpool(self):
         return resources, active
 
 
+    # --------------------------------------------------------------------------
+    #
+    def _fail_task(self, task, e, detail):
+
+        task['control']          = 'tmgr_pending'
+        task['exception']        = repr(e)
+        task['exception_detail'] = detail
+        task['target_state']     = 'FAILED'
+        task['$all']             = True
+
+        self._log.exception('scheduling failed for %s', task['uid'])
+
+        self.advance(task, rps.FAILED, publish=True, push=False,
+                           fwd=True)
```

Comment:
btw, for `AgentComponent`s we always have `fwd=True`, unless it is set explicitly `fwd=False`